### PR TITLE
Part8

### PR DESCRIPTION
**Pull request by austin-lolli**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -290,111 +304,32 @@ struct Numeric
                     return *this; 
                 }
             }
-            else if ( y < std::numeric_limits<Type>::epsilon() )
+            else if ( y < std::numeric_limits<T>::epsilon() )
```
Comment (matkatmusic) [inline main.cpp]: Either static_cast `y` to `T` or change `T` to `OtherType` to clear the warning:
```
main.cpp:314:19: warning: implicit conversion increases floating-point precision:
      'const float' to 'double' [-Wdouble-promotion]
        else if ( y < std::numeric_limits<T>::epsilon() )
                  ^ ~
main.cpp:375:7: note: in instantiation of function template specialization
      'Numeric<double>::operator/=<float>' requested here
    d /= 2.f;
```

File: main.cpp [hunk 2]
```diff
@@ -290,111 +304,32 @@ struct Numeric
...
             {
                 std::cout << "Can't divide by non-int 0! Cancelling operation." << std::endl;
                 return *this;
             }
                 
         }
-        else if ( y < std::numeric_limits<Type>::epsilon() )
+        else if ( y < std::numeric_limits<T>::epsilon() )
```
Comment (matkatmusic) [inline main.cpp]: see above comment

File: main.cpp [hunk 3]
```diff
@@ -422,126 +349,91 @@ void triple(std::unique_ptr<T> &heapNumber)
     *heapNumber *= 3;
 }
 
+template<typename T>
+void cube(std::unique_ptr<T> &heapNumber)
+{
+    // previously *heapNumber *= ( (*heapNumber) * (*heapNumber) ) but this doesnt create a temporary for assignment 
+    *heapNumber = ( (*heapNumber) * (*heapNumber) * (*heapNumber) );
+}
+
```
Comment (matkatmusic) [inline main.cpp]: All of those `*` are hard to look at.  This is much friendlier:
`auto& v = *heapNumber;   v = v * v * v;`
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp (hunk 3 above)
Comment (austin-lolli) [reply main.cpp]: That is much cleaner, thanks!

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**